sdk/core/Azure.Core/api/Azure.Core.net461.cs

```diff
@@ -120,6 +120,12 @@ public partial class MatchConditions
         public Azure.ETag? IfMatch { get { throw null; } set { } }
         public Azure.ETag? IfNoneMatch { get { throw null; } set { } }
     }
+    public static partial class ModelSerializer
+    {
+        public static T Deserialize<T>(System.IO.Stream stream, Azure.SerializableOptions? options = null) where T : Azure.IJsonSerializable, new() { throw null; }
+        public static T Deserialize<T>(string json, Azure.SerializableOptions? options = null) where T : Azure.IJsonSerializable, new() { throw null; }
```

Comment:
Would this eventually support the full set of types JsonSerializer supports?  ReadOnlySpan\<byte\> could be useful for high-performance scenarios.

Would this also be the way we would convert DynamicData to model types and vice versa?

Reply:
Yeah the overloads will definitely get filled out if we decide to go this way, only implementing some basic scenarios for now to minimize throw away.